src/feature/client/entrynodes.c

```diff
@@ -3042,6 +3073,26 @@ entry_guard_parse_from_state(const char *s)
     }
   }
 
+  if (sampled_idx) {
+    int ok = 1;
+    long idx = tor_parse_long(sampled_idx, 10, 0, INT_MAX, &ok, NULL);
+    if (!ok) {
+      log_warn(LD_GUARD, "Guard has invalid sampled_idx %s",
+          escaped(sampled_idx));
```

Comment:
Can we set the sampled_idx to _something_ in this case?  Right now it seems that we're leaving it unassigned at zero, which is not so great.

Reply:
Ok, setting missing idx starting from the max sample size to avoid those nodes to be selected before any other.